src/ray/gcs/gcs_client/gcs_client.h

```diff
@@ -241,6 +264,26 @@ class RAY_EXPORT GcsClient : public std::enable_shared_from_this<GcsClient> {
 
   /// Retry interval to reconnect to a GCS server.
   const int64_t kGCSReconnectionRetryIntervalMs = 1000;
+
+  // Protect the `reconnect_in_progress_` variable.
+  mutable absl::Mutex reconnect_flag_mutex_;
```

Comment:
Why do we need mutex when `reconnect_in_progress_ ` is already atomic?

Reply:
We need mutex, atomic seems unnecessary now.